### PR TITLE
#767 Add spring test module to facilitate Spring Tests with random MockServer port 

### PR DESCRIPTION
**Pull request by masooh**
Here are the copy of our files. But tests are still missing. I try to provide them in the next week.

### CODE REVIEW

File: mockserver-spring/pom.xml [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+        <dependency>
+            <groupId>org.springframework</groupId>
+            <artifactId>spring-context</artifactId>
+            <version>${spring.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>org.springframework</groupId>
+            <artifactId>spring-test</artifactId>
+            <version>${spring.version}</version>
+            <scope>compile</scope>
```
Comment (leonard84) [inline mockserver-spring/pom.xml:41]: Should this be `compile` or `provided`?

File: mockserver-spring/pom.xml [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    <name>MockServer Spring</name>
+    <description>A module containing helpers for writing mockserver tests in spring projects</description>
+    <url>http://www.mock-server.com</url>
+
+    <properties>
+        <spring-boot.version>2.2.8.RELEASE</spring-boot.version>
+    </properties>
+
+    <dependencies>
+        <dependency>
```
Comment (leonard84) [inline mockserver-spring/pom.xml:22]: Versions should be manged in the parent pom

File: mockserver-spring/pom.xml [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+    <modelVersion>4.0.0</modelVersion>
+
+    <artifactId>mockserver-spring</artifactId>
+    <name>MockServer Spring</name>
+    <description>A module containing helpers for writing mockserver tests in spring projects</description>
+    <url>http://www.mock-server.com</url>
+
+    <properties>
+        <spring-boot.version>2.2.8.RELEASE</spring-boot.version>
```
Comment (leonard84) [inline mockserver-spring/pom.xml:18]: Unused?

File: mockserver-spring/src/main/java/org/mockserver/spring/MockServerPropertyCustomizer.java [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    MockServerPropertyCustomizer(String... properties) {
+        this.properties = Arrays.asList(properties);
+    }
+    @Override
+    public void customizeContext(ConfigurableApplicationContext context, MergedContextConfiguration mergedConfig) {
+        context.getEnvironment().getPropertySources().addLast(
+                new MockPropertySource().withProperty("mockServerPort", mockServerPort)
+        );
+
+        if (!properties.isEmpty()) {
```
Comment (leonard84) [inline mockserver-spring/src/main/java/org/mockserver/spring/MockServerPropertyCustomizer.java:30]: isn't this check redundant?

File: mockserver-spring/src/main/java/org/mockserver/spring/MockServerTestCustomizerFactory.java [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+import org.springframework.core.annotation.AnnotatedElementUtils;
+import org.springframework.test.context.ContextConfigurationAttributes;
+import org.springframework.test.context.ContextCustomizer;
+import org.springframework.test.context.ContextCustomizerFactory;
+
+public class MockServerTestCustomizerFactory implements ContextCustomizerFactory {
+    @Override
+    public ContextCustomizer createContextCustomizer(Class<?> testClass, List<ContextConfigurationAttributes> configAttributes) {
+        final MockServerTest mockServerTest = AnnotatedElementUtils.findMergedAnnotation(testClass, MockServerTest.class);
+        if (mockServerTest != null) {
```
Comment (leonard84) [inline mockserver-spring/src/main/java/org/mockserver/spring/MockServerTestCustomizerFactory.java:14]: You could do
`return mockServerTest == null ? null : new MockServerPropertyCustomizer(mockServerTest.value());`

File: mockserver-spring/src/main/java/org/mockserver/spring/MockServerTestExecutionListener.java [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+
+import org.mockserver.client.MockServerClient;
+import org.mockserver.integration.ClientAndServer;
+import org.springframework.core.annotation.AnnotatedElementUtils;
+import org.springframework.core.env.Environment;
+import org.springframework.test.context.TestContext;
+import org.springframework.test.context.support.AbstractTestExecutionListener;
+
+@SuppressWarnings({
+        "squid:S1523" /*   Make sure that this dynamic injection or execution of code is safe.
```
Comment (leonard84) [inline mockserver-spring/src/main/java/org/mockserver/spring/MockServerTestExecutionListener.java:17]: Still relevant here?

File: mockserver-spring/pom.xml (hunk 1 above)
Comment (masooh) [reply mockserver-spring/pom.xml:41]: has to be compile due to `import org.springframework.test.context.ContextCustomizer`

File: mockserver-spring/pom.xml (hunk 3 above)
Comment (masooh) [reply mockserver-spring/pom.xml:18]: done

File: mockserver-spring/pom.xml (hunk 2 above)
Comment (masooh) [reply mockserver-spring/pom.xml:22]: done

File: mockserver-spring/src/main/java/org/mockserver/spring/MockServerPropertyCustomizer.java (hunk 4 above)
Comment (masooh) [reply mockserver-spring/src/main/java/org/mockserver/spring/MockServerPropertyCustomizer.java:30]: it is

File: mockserver-spring/src/main/java/org/mockserver/spring/MockServerTestCustomizerFactory.java (hunk 5 above)
Comment (masooh) [reply mockserver-spring/src/main/java/org/mockserver/spring/MockServerTestCustomizerFactory.java:14]: done

File: mockserver-spring/src/main/java/org/mockserver/spring/MockServerTestExecutionListener.java (hunk 6 above)
Comment (masooh) [reply mockserver-spring/src/main/java/org/mockserver/spring/MockServerTestExecutionListener.java:17]: probably not, we'll see

**Pull request merged**